### PR TITLE
pythonPackage.jsonlines: init at 1.2.0

### PR DESCRIPTION
**Pull request by sondr3**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/jsonlines/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    inherit pname version;
+    sha256 = "126yxzxq7yqhkd3gxmksxrd5kv10iviq15d4lk464j4xi9cdbf23";
+  };
+
+  propagatedBuildInputs = [ six ];
+
+  checkInputs = [ tox flake8 pep8-naming pytest pytestcov pytestpep8 ];
+  doCheck = false; # tox fails
+  checkPhase = ''
+    tox
```
Comment (jonringer) [inline pkgs/development/python-modules/jsonlines/default.nix]:
```suggestion
    pytest
```

**Review by jonringer — COMMENTED** [review]
tox is for emulating different interpreters, nix already parameterizes this.

File: pkgs/development/python-modules/jsonlines/default.nix (hunk 1 above)
Comment (sondr3) [reply pkgs/development/python-modules/jsonlines/default.nix]: Thanks, a little progress. Now it instead finds no tests and exits with `code 5`:

```
running install tests
/nix/store/afxxp90mqz8mhxxyvf9fjgkx8wx6m18a-python3.7-pep8-1.7.1/lib/python3.7/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
============================= test session starts ==============================
platform linux -- Python 3.7.4, pytest-5.1.0, py-1.8.0, pluggy-0.12.0
rootdir: /build/jsonlines-1.2.0
plugins: cov-2.6.1, pep8-1.0.6
collected 0 items                                                              

============================ no tests ran in 0.01s =============================
builder for '/nix/store/ip3spm2r9bqhg07amb19rasn312kn5ag-python3.7-jsonlines-1.2.0.drv' failed with exit code 5
error: build of '/nix/store/ip3spm2r9bqhg07amb19rasn312kn5ag-python3.7-jsonlines-1.2.0.drv' failed
```

Comment (sondr3) [reply pkgs/development/python-modules/jsonlines/default.nix]: Oh wait, I see why, using `pypi` doesn't include the tests in their release. I'll use GitHub instead.

Comment (sondr3) [reply pkgs/development/python-modules/jsonlines/default.nix]: Done, I've updated the release @jonringer

File: pkgs/development/python-modules/jsonlines/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "wbolster";
+    repo = pname;
+    rev = version;
+    sha256 = "1f8zsqy8p9a41gqg2a5x7sppc5qhhq7gw58id2aigb270yxzs7jw";
+  };
+
+  propagatedBuildInputs = [ six ];
+
+  checkInputs = [ tox flake8 pep8-naming pytest pytestcov pytestpep8 ];
```
Comment (jonringer) [inline pkgs/development/python-modules/jsonlines/default.nix]:
```suggestion
  checkInputs = [ flake8 pep8-naming pytest pytestcov pytestpep8 ];
```

File: pkgs/development/python-modules/jsonlines/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ lib, fetchFromGitHub, buildPythonPackage, six
+, flake8, pep8-naming, pytest, pytestcov, pytestpep8, tox }:
```
Comment (jonringer) [inline pkgs/development/python-modules/jsonlines/default.nix]:
```suggestion
, flake8, pep8-naming, pytest, pytestcov, pytestpep8 }:
```

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS (python38Packages.pytest is broken on master)
diff LGTM
tests run

```
[4 built (1 failed), 105 copied (47.0 MiB), 13.8 MiB DL]
error: build of '/nix/store/57d9l07s5yks5anwr9w4nxd859cb8crm-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/71705
1 package failed to build:
python38Packages.jsonlines

2 package were build:
python27Packages.jsonlines python37Packages.jsonlines
```

**Comment by jonringer** [comment]
please cleanup tox, and then it LGTM

**Comment by sondr3** [comment]
Done :smiley:

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.jsonlines python37Packages.jsonlines

**Comment by sondr3** [comment]
ping @jonringer :sweat_smile:

**Pull request merged**